Environment variables docs

schottj opened the pull request:
## Description & motivation
Adding documentation for environment variables and private packages



## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [X] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`

Commit:
update the adapter page about dbt init

barberscott committed:
Update Slim CI docs with troubleshooting for "defer to self" case

It is possible for a customer to set Slim CI jobs to defer to SELF -- in doing so, this will produce an error message that references the immediate prior PR.  Adding details on troubleshooting and fixing this issue.

samkessaram committed:
Fix docs grammar

Commit:
update section to say other files and include version file

Commit:
include mention of create adapter script

Commit:
1.1.32 changelog

schottj committed:
Update dbt-cloud-changelog.md

Interim edits

schottj committed:
Update dbt-cloud-changelog.md

LGTM

amychen1776 committed:
Update website/docs/docs/contributing/building-a-new-adapter.md

Co-authored-by: Jeremy Cohen <jeremy@fishtownanalytics.com>

Commit:
Dashboard Status Tile Information

barryaron committed:
Update website/docs/docs/dbt-cloud/using-dbt-cloud/cloud-dashboard-status-tiles.md

Co-authored-by: Drew Banin <drew@fishtownanalytics.com>

barryaron committed:
Update cloud-dashboard-status-tiles.md

barryaron committed:
Update cloud-dashboard-status-tiles.md

amy-byrum committed:
Update setting-up-sso-with-saml-2.0.md

Changed the logo download links to point to the new assets

amy-byrum committed:
Update setting-up-sso-with-saml-2.0.md

Changed the link to just point to the logo file instead of the whole folder

lfracalossiFA committed:
Updating image for social sharing meta

hanianiani committed:
correct small typo

change "can also includes" to "can also include" :)

Sagnik-Chaudhuri committed:
Update 2b-create-a-project-dbt-cli.md

fixed default project name, changed from my_new_package to my_new_project

Sagnik-Chaudhuri committed:
Update 2b-create-a-project-dbt-cli.md

fixed default project name, changed from my_new_package to my_new_project

amychen1776 committed:
Update the query tag section

Commit:
Update dbt-cloud-changelog.md

schottj committed:
Interim changees

schottj committed:
Update dbt-cloud-changelog.md

LGTM

schottj committed:
Small update

cesher committed:
Update docs website with API v4

schottj committed:
Update understanding-state.md

Removing the beta feature call out because we recently were asked if SlimCI is still in beta

schottj committed:
Update cloud-snapshotting-source-freshness.md

Do we need a cloud call out here? This whole section is about cloud, so I found it strange for a separate cloud call out.

jschottenstein committed:
first cut of environment variables docs

drewbanin commented on website/docs/docs/building-a-dbt-project/package-management.md:
```diff
@@ -108,33 +108,78 @@ To find the latest release for a package, navigate to the `Releases` tab in the
+### Private packages
 
+#### SSH Key Method
+Private packages can be installed by using the SSH configuration on the machine. This method requires the SSH configuration be stored in `~/.ssh/`.
```
This reads a little funny - is this a place where it makes sense to link out to docs like:
 - https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh
 - https://docs.gitlab.com/ee/ssh/

?

schottj replied:
Sounds good I'll update it. I tried to preserve the original language used in the docs, but it wasn't that clear to me either.

drewbanin commented on website/docs/docs/dbt-cloud/using-dbt-cloud/cloud-environment-variables.md:
```diff
@@ -3,23 +3,79 @@ title: Environment variables
+[TO CREATE GIF]
+
+:::info Appropriate coverage
+If you have not set a project level default value for every environment variable, it may be possible that dbt Cloud does not know how to interpret the value of an environment variable in all contexts. We leave it to our users to either supply a default or make sure all scenarios where dbt runs have been covered; otherwise, dbt will throw a compile error.
```
```suggestion
If you have not set a project level default value for every environment variable, it may be possible that dbt Cloud does not know how to interpret the value of an environment variable in all contexts. You should either supply a default value or ensure that specific environment variables are specified for all relevant environments; otherwise, dbt will throw a compile error.
```

drewbanin replied:
I don't feel *incredibly* attached to this specific wording, but I think we should avoid talking about "we" as the people who make dbt Cloud and we should definitely avoid using the possessive "we" when talking about dbt Cloud users.

drewbanin commented on website/docs/docs/dbt-cloud/using-dbt-cloud/cloud-environment-variables.md:
```diff
@@ -3,23 +3,79 @@ title: Environment variables
+
+### Handling secrets
+
+While all environment variables are encrypted in our database regardless of whether they are secrets or not, we want to apply extra care when using environment variables as secrets. If you want a particular environment variable to be scrubbed from all logs and error messages, in addition to obfuscating the value in the UI, you need to prefix the key with `DBT_ENV_SECRET_`. This functionality is s ...
```
```suggestion
While all environment variables are encrypted at rest in dbt Cloud, dbt Cloud has additional capabilities for managing environment variables with secret or otherwise sensitive values. If you want a particular environment variable to be scrubbed from all logs and error messages, in addition to obfuscating the value in the UI, you can prefix the key with `DBT_ENV_SECRET_`. This functionality is supported from `dbt v0.21.0` and on. 
```

drewbanin commented on website/docs/docs/dbt-cloud/using-dbt-cloud/cloud-environment-variables.md:
```diff
@@ -29,16 +85,22 @@ PR builds triggered via a webhook_
+#### Clone private packages
+Now that you can set secrets as environment variables, you can pass git tokens into your package HTTPS URLs to allow for on-the-fly cloning of private repositories. Read more about enabling [private package cloning](/building-a-dbt-project/package-management#private-packages).
+#### Dynamically set your warehouse in your Snowflake connection
+Environment variables can make overriding warehouses possible.  
```
```suggestion
Environment variables can make it simpler to override a Snowflake virtual warehouse for specific dbt Cloud jobs or environments.  
```

drewbanin commented on website/docs/docs/dbt-cloud/using-dbt-cloud/cloud-environment-variables.md:
```diff
@@ -29,16 +85,22 @@ PR builds triggered via a webhook_
+#### Dynamically set your warehouse in your Snowflake connection
+Environment variables can make overriding warehouses possible.  
+
+Suppose you'd like to set your warehouse in Snowflake depending on the job. You'd like to run a full-refresh job in a XL warehouse, but your incremental job only needs to run in a medium-sized warehouse. Both jobs run in the same dbt environment. In your connection configuration, you should use an environment variable to set the warehouse name to `{{env_var('WAREHOUSE')}}`. Then depending on the  ...
```
```suggestion
Suppose you'd like to set your warehouse in Snowflake to different values for different jobs. You'd like to run a full-refresh job in an XL warehouse, but your incremental job only needs to run in a medium-sized warehouse. Both jobs are configured in the same dbt Cloud environment. In your connection configuration, you can use an environment variable to set the warehouse name to `{{env_var('WAREHOUSE')}}`. Then in the job configuration, different values for the `WAREHOUSE` environment variable can be specified to dynamically configure different warehouses for each workload.
```

drewbanin reviewed (commented):
Looks like this is still in progress, but wanted to add some early feedback! Let me know if you have any thoughts or questions :)

schottj committed:
Apply suggestions from code review

These all look good to me!

Co-authored-by: Drew Banin <drew@fishtownanalytics.com>

schottj committed:
Update cloud-enabling-continuous-integration-with-github.md

We have problems with CI jobs that defer to themselves, so I want to remove this from the docs

JKarlavige committed:
fixes nav breakpoint issue

jschottenstein committed:
changed ssh wording

jschottenstein committed:
added pictures

drewbanin commented on website/docs/docs/building-a-dbt-project/package-management.md:
```diff
@@ -108,33 +108,80 @@ To find the latest release for a package, navigate to the `Releases` tab in the
+### Private packages
+
+#### SSH Key Method
+Private packages can be installed via SSH keys, which authenticate to remote servers. You'll need to add your key to your git provider and update your ~/.ssh/config file to manage your settings.
```
```suggestion
Private packages can be installed via SSH keys, which authenticate to remote git servers. You'll need to add your key to your git provider.
```

drewbanin replied:
this is still _kind of_ funny - you usually don't need to touch `~/.ssh/config` when setting up a git connection over SSH. I think we already litigated these lines and they're not wrong... I think it might just lead to some questions!

If you're unhappy with them, we can spend some more time workshopping this together. If you think it's ok, then I don't find them offensive enough to protest :)

drewbanin reviewed (commented):
This LGTM with one small and not particularly significant comment. It does look like some commits are showing up that _should not_ be in this PR -- might be worth rebasing against `current` to keep the diff in this PR to only the changes relevant to environment variables. Let me know if I can lend a hand with that!

jschottenstein committed:
trying to merge

schottj committed:
Update website/docs/docs/building-a-dbt-project/package-management.md

small change

Co-authored-by: Drew Banin <drew@fishtownanalytics.com>
